backend/logic/courses/courses.js

```diff
@@ -1,19 +1,72 @@
-const mysql = require('../../sql/connection');
+const mysql = require("../../sql/connection");
+const format = require("../../validation/format")
 
-async function queryCourseByTag(tag) {
-    if (!tag) {
-        throw Error("Undefined tag")
-    }
+var queryCourseByTag = async function queryCourseByTag(tag) {
+  if (!tag) {
+    throw Error("Undefined tag")
+  }
+
+  let connection = await mysql.getNewConnection();
+  try {
+    let courses = await connection.query("SELECT course_code FROM course_tag WHERE course_tag.tag_name LIKE ?", tag);
+    connection.release();
+    return (JSON.parse(JSON.stringify(courses)));
+  } catch (error) {
+    connection.release();
+    console.error(error);
+    throw Error(error.message);
+  }
+}
+
+var addCompletedCourses = async (studentId, courses, section) => {
+  if (!format.verifyStudentId(studentId)) {
```

Comment:
i believe the verifyStudentId method should throw the error itself, less code repeat that way

Reply:
Will be handled in a separate commit